WordPress/gutenberg
# Implement image alignment controls

This pull request seeks to implement alignment controls for the image block.

![image alignment](https://cloud.githubusercontent.com/assets/1779930/24979192/51ee4d14-1fa1-11e7-88e7-f89badfcb0f1.png)

__Implementation notes:__

- `Image` was extracted to a top-of-file common control to more easily maintain common behaviors between `edit` and `save` variations
- Parsing `align` from the `className` is a bit ugly with the current setup. I'm very inclined to change this to a comment-serialized attribute and support legacy class-based alignment with a compatibility layer like the one explored in #413
- Styling is very ugly when performing the alignment

__Testing instructions:__

1. Navigate to the Gutenberg admin screen.
2. Click the first image
3. Note a toolbar with alignment controls is shown
4. Note the "No alignment" button is active
5. Click an alignment button
6. Note the alignment of the image changes accordingly
7. Note the clicked alignment button is now active
8. Click the same button
9. Note that alignment resets to "No alignment"
